packages/kbn-visualization-ui-components/components/field_picker/field_picker.tsx

```diff
@@ -32,14 +32,21 @@ export function FieldPicker<T extends FieldOptionValue = FieldOptionValue>(
   props: FieldPickerProps<T>
 ) {
   const {
-    selectedOptions,
+    activeField,
     options,
     onChoose,
     onDelete,
     fieldIsInvalid,
     ['data-test-subj']: dataTestSub,
     ...rest
   } = props;
+
+  const [selectedOptions, setSelectedOptions] = React.useState(activeField ? [activeField] : []);
+
+  React.useEffect(() => {
+    setSelectedOptions(activeField ? [activeField] : []);
+  }, [activeField]);
```

Comment:
I'm not sure if this is a good practice (the use of useEffect to update a subsequent change of the activeField from the props), but I don't have a solution to that, worth checking the react guidelines on that https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
Solved it!